evm-contracts/src/v0.7/dev/CompoundPriceFlaggingValidator.sol

```diff
@@ -0,0 +1,277 @@
+    string symbol;
+    // Used to convert price to match aggregator decimals
+    uint8 decimals;
+    // 1        = 100%
+    // 5        = 20%
+    // 10       = 10%
+    // 20       = 5%
+    // 50       = 2%
+    // 100      = 1%
+    // 200      = 0.5%
+    // 500      = 0.2%
+    // 1000     = 0.1%
+    uint32 deviationThresholdDenominator;
+  }
+
+  FlagsInterface private s_flags;
+  UniswapAnchoredView private s_compOpenOracle;
+  mapping(address => CompoundFeedDetails) private s_thresholds;
+
+  event CompoundOpenOracleAddressUpdated(
+    address indexed from,
+    address indexed to
+  );
+  event FlagsAddressUpdated(
+    address indexed from,
+    address indexed to
+  );
+  event ThresholdUpdated(
+    address indexed aggregator,
+    string symbol,
+    uint8 decimals,
+    uint32 deviationThresholdDenominator
+  );
+  
+  /**
+   * @notice Create a new CompoundPriceFlaggingValidator
+   * @dev Use this contract to compare Chainlink aggregator prices
+   * against the Compound Open Oracle prices
+   * @param flagsAddress Address of the flag contract
+   * @param compoundOracleAddress Address of the Compound Open Oracle UniswapAnchoredView contract
+   */
+  constructor(address flagsAddress, address compoundOracleAddress)
+    ConfirmedOwner(msg.sender)
+  {
+    setFlagsAddress(flagsAddress);
+    setCompoundOpenOracleAddress(compoundOracleAddress);
+  }
+
+  /**
+   * @notice Set the address of the Compound Open Oracle UniswapAnchoredView contract
+   * @param oracleAddress Compound Open Oracle UniswapAnchoredView address
+   */
+  function setCompoundOpenOracleAddress(address oracleAddress)
+    public
+    onlyOwner()
+  {
+    address previous = address(s_compOpenOracle);
+    if (previous != oracleAddress) {
+      s_compOpenOracle = UniswapAnchoredView(oracleAddress);
+      emit CompoundOpenOracleAddressUpdated(previous, oracleAddress);
+    }
+  }
+
+  /**
+   * @notice Updates the flagging contract address for raising flags
+   * @param flagsAddress sets the address of the flags contract
+   */
+  function setFlagsAddress(address flagsAddress)
+    public
+    onlyOwner()
+  {
+    address previous = address(s_flags);
+    if (previous != flagsAddress) {
+      s_flags = FlagsInterface(flagsAddress);
+      emit FlagsAddressUpdated(previous, flagsAddress);
+    }
+  }
+
+  /**
+   * @notice Set the threshold details for comparing a Chainlink aggregator
+   * to a Compound Open Oracle feed.
+   * @param aggregator The Chainlink aggregator address
+   * @param compoundSymbol The symbol used by Compound for this feed
+   * @param compoundDecimals The number of decimals in the Compound feed
+   * @param compoundDeviationThresholdDenominator The threshold denominator use to determine
+   * the percentage with which the difference in prices must reside within. For example:
+   * If the prices are valid within a 5% threshold, and 1 / denominator, this denominator
+   * = 20, since 1/20 = 5.
+   */
+  function setThreshold(
```

Comment:
Should we validate that `deviationThresholdDenominator` isn't 0? Any other validations we could do? Would be nice to check that `compoundSymbol` actually has a price on the compound oracle.